cats/cats-core/src/main/java/com/netflix/spinnaker/cats/agent/DefaultAgentScheduler.java

```diff
@@ -63,8 +63,15 @@ public DefaultAgentScheduler(ScheduledExecutorService scheduledExecutorService,
 
     @Override
     public void schedule(Agent agent, AgentExecution agentExecution, ExecutionInstrumentation executionInstrumentation) {
+        Long agentInterval = interval;
+        TimeUnit agentTimeUnit = timeUnit;
+        if (agent instanceof AgentIntervalAware) {
+          agentInterval = ((AgentIntervalAware) agent).getAgentInterval();
+          agentTimeUnit = TimeUnit.MILLISECONDS;
+        }
+
         Future agentFuture =
-          scheduledExecutorService.scheduleAtFixedRate(new AgentExecutionRunnable(agent, agentExecution, executionInstrumentation), 0, interval, timeUnit);
+          scheduledExecutorService.scheduleAtFixedRate(new AgentExecutionRunnable(agent, agentExecution, executionInstrumentation), 0, agentInterval, agentTimeUnit);
```

Comment:
I think this needs to be done in other places IIRC... weird that this is taking a Long and not an Interval object, because it'll need some way to get the interval's timeout

Reply:
See above, I think by modifying the `DefaultAgentIntervalProvider` we set set the agent intervals properly with this new configuration.

Reply:
Sounds good